src/js/services/quiz.service.ts

```diff
@@ -8,17 +8,51 @@ import { Observable } from 'rxjs/Observable';
 import { PlaylistService } from './playlist.service';
 import { QuestionsComponent } from '../components/questions/questions.component';
 
+import { Track } from '../models/track';
+
 @Injectable()
 export class QuizService {
 
   private _questionsComponent: QuestionsComponent;
   private _totalQuestions = 0;
+  private _tracks: Track[];
 
   constructor(private playlistService: PlaylistService) {
     this.playlistService.getPlaylist().subscribe(
       playlist => this._totalQuestions = 5,
       error => console.error(error)
       );
+
+    if (this._totalQuestions > this.playlistService.tracks.length) {
+      this._totalQuestions = this.playlistService.tracks.length;
+    }
+    this._tracks = this.getRandomTracks(this._totalQuestions);
```

Comment:
`console.log(this._tracks)` currently gives me an empty array.
Probably an issue due to async calls. Needs a short check here - if anyone's got time fo dat.

![Ain't nobody got time fo dat](https://media.giphy.com/media/ILmqMop0Q0xsQ/giphy.gif)